include/simdjson/common_defs.h

```diff
@@ -121,10 +121,29 @@ constexpr size_t DEFAULT_MAX_DEPTH = 1024;
 
 #endif // MSC_VER
 
+
+// some systems have string_view even if SIMDJSON_CPLUSPLUS17 is 0,
+// let us try to detect them:
+#if defined __has_include
+// do not combine the next #if with the previous one (unsafe)
+#if __has_include (<string_view>)
+// now it is safe to trigger the include
+#include <string_view> // though the file is there, it does not follow that we got the implementation
+#if defined(_LIBCPP_STRING_VIEW)
+// Ah! So we under libc++ which under its Library Fundamentals Technical Specification, which preceeded C++17,
+// included string_view.
+// This means that we have string_view *even though* we may not have C++17.
+#define SIMDJSON_HAS_STRING_VIEW
+#endif// _LIBCPP_STRING_VIEW
```

Comment:
Any chance of putting a space after endif and before //? Sorry, programmer OCD. (Also happy to change them all, I'm just unusually sensitive to these things.)

Reply:
Done.